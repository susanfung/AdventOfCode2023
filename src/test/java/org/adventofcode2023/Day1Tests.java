package org.adventofcode2023;


import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import static org.adventofcode2023.Day1.*;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class Day1Tests
{
  @Test
  public void testFindFirstDigit()
  {
      String[] findFirstDigitResults = {findFirstDigit("1abc2"), findFirstDigit("pqr3stu8vwx"), findFirstDigit(
              "a1b2c3d4e5f"), findFirstDigit("treb7uchet")};
      Approvals.verifyAll("First Digit", findFirstDigitResults);
  }

  @Test
  public void testFindLastDigit()
  {
      String[] findLastDigitResults = {findLastDigit("1abc2"), findLastDigit("pqr3stu8vwx"), findLastDigit(
              "a1b2c3d4e5f"), findLastDigit("treb7uchet")};
      Approvals.verifyAll("Last Digit", findLastDigitResults);
  }
}
