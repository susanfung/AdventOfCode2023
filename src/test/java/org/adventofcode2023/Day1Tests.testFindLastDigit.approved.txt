Last Digit[0] = 2
Last Digit[1] = 8
Last Digit[2] = 5
Last Digit[3] = 7
